compiler/next/include/chpl/AST/Location.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_AST_LOCATION_H
+#define CHPL_AST_LOCATION_H
+
+#include "chpl/AST/UniqueString.h"
+
+namespace chpl {
+namespace ast {
+
+/**
+  This class represents a source location.
+ */
+class Location final {
+ private:
+  UniqueString path_;
+  int firstLine_;
+  int firstColumn_;
+  int lastLine_;
+  int lastColumn_;
+ public:
+  Location()
+    : path_(),
+      firstLine_(-1), firstColumn_(-1),
+      lastLine_(-1), lastColumn_(-1) {
+  }
+
+  Location(UniqueString path,
+           int firstLine=-1, int firstColumn=-1,
+           int lastLine=-1, int lastColumn=-1)
+    : path_(path),
+      firstLine_(firstLine), firstColumn_(firstColumn),
+      lastLine_(lastLine), lastColumn_(lastColumn) {
+  }
+
+  UniqueString path() { return path_; }
+
+  inline bool operator==(const Location other) const {
+    return this->path_ == other.path_ &&
+           this->firstLine_ == other.firstLine_ &&
+           this->firstColumn_ == other.firstColumn_ &&
+           this->lastLine_ == other.lastLine_ &&
+           this->lastColumn_ == other.lastColumn_;
+  }
+  inline bool operator!=(const Location other) const {
```

Comment:
For brevity and consistency, I usually only define `operator==` and `operator<`, and then all other relational operators can be defined based on those, such as `operator!=(a,b) { return !(a==b); }`